ai_services/anomaly_detection/data_preprocessing_examples/oci_data_flow_based_examples/example_code/df_driver.py

```diff
@@ -0,0 +1,167 @@
+import oci
+import json
+import argparse
+import pandas as pd
+
+from pyspark.sql import SparkSession
+from pyspark.sql import functions as F
+from io import StringIO
+from example_code.remove_unnecessary_columns import remove_unnecessary_columns
+from example_code.column_rename import column_rename
+from example_code.one_hot_encoding import one_hot_encoding
+from example_code.string_transformations import string_transformation
+from example_code.format_timestamp import format_timestamp
+from example_code.temporal_differencing import temporal_differencing
+from example_code.normalization import normalize_data
+from example_code.fixed_window_batching import windowing
+from example_code.sliding_window_aggregation import aggregation
+from example_code.one_hot_encoding import one_hot_encoding
+from example_code.pivoting import spark_pivoting
+from example_code.sharding import sharding
+
+from example_code.time_series_merge import time_series_merge
+from example_code.time_series_join import time_series_join
+
+signer = oci.auth.signers.get_resource_principals_signer()
+data_flow_client = oci.data_flow.DataFlowClient(config={}, signer=signer)
+object_storage_client = oci.object_storage.ObjectStorageClient(config={}, signer=signer)
+SINGLE_DATAFRAME_PROCESSING = "singleDataFrameProcessing"
+COMBINE_DATAFRAMES = "combineDataFrames"
+JOIN = "join"
+MERGE = "merge"
+RESERVED_METADATA = ['distinct_categories']
+
+def get_token_path(spark):
+    token_key = "spark.hadoop.fs.oci.client.auth.delegationTokenPath"
+    token_path = spark.sparkContext.getConf().get(token_key)
+    return token_path
+
+def get_authenticated_client(token_path, client):
+    if token_path is None:
+        # You are running locally, so use our API Key.
+        config = oci.config.from_file()
```

Comment:
actually, let's read from local filesystem instead of obj storage in local mode. that way we won't need the user to set up oci config etc. This can come in the next iteration

Reply:
[Minor] Also, please support optional profile name (sample [snippet](https://github.com/sudharkj/oci-data-science-ai-samples/commit/bedc36f312f3db5990f910d4c93e3b8561412d7a#diff-01004b5cf83417c2b0c1f208746048b2d4931e54ce5b7b55988ba76c285cd439R27-R33))

Reply:
Added TODO list